frontends/p4/unusedDeclarations.h

```diff
@@ -70,14 +94,22 @@ class RemoveUnusedDeclarations : public Transform {
     const IR::Node* preorder(IR::Type_Declaration* decl) override { return process(decl); }
 };
 
-/// Iterates RemoveUnusedDeclarations until convergence.
+/** @brief Iterates RemoveUnusedDeclarations until convergence.
+ *
+ * If @warn is true, emit compiler warnings if an unused instance of an
+ * IR::P4Table or IR::Declaration_Instance is removed.
+ */
 class RemoveAllUnusedDeclarations : public PassManager {
  public:
     explicit RemoveAllUnusedDeclarations(ReferenceMap* refMap, bool warn = false) {
         CHECK_NULL(refMap);
         std::set<const IR::Node*> *warned = nullptr;
+
+        // TODO: The @warned set is discarded.  Is it necessary, or can it be replaced
```

Comment:
Oh I see.  This avoids issuing multiple warnings for the same unreferenced extern definition, which is the only thing that can both trigger a warning and persist across multiple rounds.

I'll remove the TODO and add a comment to this effect.  Thanks for pointing it out.